docs/eom_base.rst

```diff
@@ -0,0 +1,372 @@
+
+The first step is to define the dynamic and constant symbols that will be used
+in the system. ::
+
+    >>> x1, x2, u1, u2 = dynamicsymbols('x1 x2 u1 u2')
+    >>> m1, c1, k1 = symbols('m1 c1 k1')
+    >>> m2, c2, k2 = symbols('m2 c2 k2')
+
+Next step is to define the equations of motion in multiple forms:
+
+[1] x' = F(x, t, r, p)
+
+[2] M(x, p) x' = F(x, t, r, p)
+
+[3] M(q, p) u' = F(q, u, t, r, p)
+    q' = G(q, u, t, r, p) ::
+
+    >>> mm = Matrix([[m1,  0],
+    ...             [0,  m2]])
+    >>> f = Matrix([-(k1+k2)*x1 + k2*x2 - (c1+c2)*u1 + c2*u2,
+    ...             k2*x1 - k2*x2 + c2*u1 - c2*u2])
+    >>> mm_full = Matrix([[1, 0,  0,  0],
+    ...                   [0, 1,  0,  0],
+    ...                   [0, 0, m1,  0],
+    ...                   [0, 0,  0, m2]])
+    >>> f_full = Matrix([u1,
+    ...                  u2,
+    ...                  -(k1+k2)*x1 + k2*x2 - (c1+c2)*u1 + c2*u2,
+    ...                  k2*x1 - k2*x2 + c2*u1 - c2*u2])
+    >>> G = Matrix([u1, u2])
+    >>> RHS = mm_full.LUsolve(f_full)
+
+Now the various forms of varible entry to `eombase` will be initiated ::
+
+    >>> coordinates = (x1, x2)
+    >>> speeds = (u1, u2)
+    >>> states = (x1, x2, u1, u2)
+
+This example will also show `system`'s ability to calculate an arbitrary user
+defined functions along with the states. Specifically the kinetic energy and
+potential energy  of the whole system is going to be determined. ::
+
+    >>> KE = 0.5 * (m1*u1**2 + m2*u2**2)
+    >>> PE = symbols("PE")
+    >>> out_eqns = {"kinetic energy": KE, PE: 0.5*k1*x1**2 + 0.5*k2*(x2-x1)**2}
+
+The `eombase` instances are now ready to be initialized. To showcase possible
+input combinations, all three of the equations of motion forms will be used. ::
+
+    >>> eom1 = eombase.EOM(coordinates, RHS, speeds=speeds, 
+    ...                    output_eqns=out_eqns)
+    >>> eom2 = eombase.EOM(states, f_full, mass_matrix=mm_full,
+    ...                    num_coordinates=2)
+    >>> eom3 = eombase.EOM(states, f, mass_matrix=mm, coordinate_derivatives=G)
+
+There are some potential problems that can occur during the initialization of
+the eombase.EOM class. First, the class assumes if it recieves
+coordinate_derivatives that the input equations of motion are in form [3]. This
+means it will produce an error if a mass matrix is not recieved. Similarly if a
+mass matrix is recieved and coordinate_derivatives are not specified, the class
+is assuming that the equations of motion have been put in in form [2]. If the
+equations of motion are input in form [1], the class will not have a notion of
+the mass matrix or the forcing vector and as such will return errors if there
+is an attempt to access these attributes. ::
+
+    >>> eom4 = eombase.EOM(states, f, coordinate_derivatives=G)
+    SyntaxError: Need to specify a mass matrix for eom form [3]
+    >>> eom1.mass_matrix
+    AttributeError: Mass matrix is not specified for eom form [1]
+    >>> eom1.mass_matrix_full
```

Comment:
For the inputs to the SymbolicSystem initialization I'm leaving mass_matrix and coordinate_derivatives to both match the current api and because mass_matrix can take in either dyn_implicit_mat or comb_implicit_mat. I'll change the attributes to match the new names.